volatility3/framework/symbols/linux/extensions/__init__.py

```diff
@@ -513,6 +608,66 @@ def get_mnt_parent(self):
     def get_mnt_mountpoint(self):
         return self.mnt_mountpoint
 
+    def get_flags_access(self) -> str:
+        return "ro" if self.get_mnt_flags() & self.MNT_READONLY else "rw"
+
+    def get_flags_opts(self) -> Iterable[str]:
+        flags = [self.MNT_FLAGS[mntflag] for mntflag in self.MNT_FLAGS if mntflag & self.get_mnt_flags()]
+        return flags
+
+    def is_shared(self) -> bool:
+        return self.get_mnt_flags() & self.MNT_SHARED
+
+    def is_unbindable(self) -> bool:
+        return self.get_mnt_flags() & self.MNT_UNBINDABLE
+
+    def is_slave(self) -> bool:
+        return self.mnt_master and self.mnt_master.vol.offset != 0
+
+    def get_devname(self) -> str:
+        return utility.pointer_to_string(self.mnt_devname, count=255)
+
+    def has_parent(self) -> bool:
+        return self.vol.offset != self.mnt_parent
+
+    def get_dominating_id(self, root) -> int:
+        """Get ID of closest dominating peer group having a representative under the given root."""
+        current_mnt = self.mnt_master
+        while current_mnt and current_mnt.vol.offset != 0:
```

Comment:
all loops like this need a 'seen' that tracks elements traversed already and breaks if duplicates are found. smear will definitely cause infinite loops eventually if not.

Reply:
done